### PR TITLE
[tests-only] Refactor public link overwrite function

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
This PR refactors `thePublicOverwritesFileWithContentUsingWebDavApi` function so that it can be used for both new and old webDavApi

## Related Issue

- Fixes <issue_link>

## Motivation and Context

I was writing some tests for `files_antivirus` https://github.com/owncloud/files_antivirus/pull/460 and I needed a step for uploading overwrite using `new` webDavApi but couldn't find one. 

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingIntUsers2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/150/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiSharePublicLink1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/77/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingPublic2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/152/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingPublic1-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/151/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-5-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/176/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProvisioningGroups-v2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/63/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareCreateSpecialToRoot1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/64/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareCreateSpecialToShares1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31781/65/1

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 1]
```diff
@@ -548,25 +548,28 @@ public function thePublicHasUploadedFileWithPasswordAndContentUsingPublicWebDAVA
...
 	 */
-	public function publiclyOverwritingContent($filename, $body = 'test') {
-		$this->publicUploadContent($filename, '', $body, false);
+	public function publiclyOverwritingContent($filename, $body = 'test', $publicWebDAVAPIVersion='old') {
+		$this->publicUploadContent($filename, '', $body, false, [], $publicWebDAVAPIVersion);
 	}
 
 	/**
-	 * @When the public overwrites file :filename with content :body using the old WebDAV API
+	 * @When /^the public overwrites file "([^"]*)" with content "([^"]*)" using the (old|new) WebDAV API$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:560]: you can use it like this. regex can be omitted
```suggestion
	 * @When the public overwrites file :filename with content :body using the :type WebDAV API
```

**Review by phil-davis — APPROVED** [review]

**Pull request merged**